src/index.ts

```diff
@@ -1,82 +1,108 @@
+  getRoles,
   generateRole,
-  generatePermissions
+  generatePermissions,
+  createUserRole,
 } from './utils/role';
 
-import { generateRowLevelSecurity } from './utils/rowlevelsecurity';
-import { generateUser } from './utils/user';
+import {
+  createRowlevelSecurity,
+  generateRowLevelSecurity,
+  getAvailableRowlevelSecurityFromSuperset,
+} from './utils/rowlevelsecurity';
+import { CSVUser, createUserAccount, generateUser } from './utils/user';
 
+import { IRowLevelSecurity } from './utils/interface';
 import {
-  postRequest,
-  getPermissionsByRoleID,
-  stringifyRequest
-} from './utils/superset';
+  addPermissionsForUserRole,
+  getUserPermissions,
+} from './utils/permissions';
 
-import { resolveUrl } from './utils/url';
+const readAndParse = async (fileName: string) => {
+  const tokens = await loginResult();
 
-const DASHBOARD_VIEWER_ROLE_ID = 9;
-const API_URL = resolveUrl(SUPERSET.baseURL, SUPERSET.apiPath);
+  const csrfToken = await getCSRFToken(tokens.bearerToken);
 
-const readAndParse = async (fileName: string) => {
-    const BEARER_TOKEN = await getBearerToken(
-      API_URL,
-      {username: SUPERSET.username, password: SUPERSET.password, provider: 'db'
-    });
-    const CSRF_TOKEN = await getCSRFToken(API_URL, BEARER_TOKEN);
-    const COOKIE = await getCookie(); // To-Do
-    const AUTHORIZATION_HEADERS = getFormattedHeaders(BEARER_TOKEN, CSRF_TOKEN, COOKIE);
-    const dashboardViewerPermissions = await getPermissionsByRoleID(
-      API_URL,
-      AUTHORIZATION_HEADERS,
-      DASHBOARD_VIEWER_ROLE_ID
-    );
-    const PERMISSIONS = dashboardViewerPermissions.result.map((item: { id: number; }) => item.id);
-    
-    fs.createReadStream(fileName, 'utf-8')
+  const headers = getFormattedHeaders(
+    tokens.bearerToken,
+    csrfToken,
+    tokens.cookie,
+  );
+
+  const rolesAvailableOnSuperset = await getRoles(headers);
+
+  const userPermissions = await getUserPermissions(
+    rolesAvailableOnSuperset,
+    headers,
+  );
+
+  const users: CSVUser[] = [];
+
+  const { result: rowLevelFromSuperset } =
+    await getAvailableRowlevelSecurityFromSuperset(headers);
+
+  fs.createReadStream(fileName, 'utf-8')
     .on('error', () => {
-        // handle error
+      // handle error
     })
     .pipe(csv())
-    .on('data', async (row: any) => {
-        const role = generateRole(row.role, row.place);
-        const rolePermissions = generatePermissions(PERMISSIONS);
-        let result = await postRequest(
-          API_URL,
-          AUTHORIZATION_HEADERS,
-          `/security/roles/`,
-          stringifyRequest(role)
+    .on('data', (data) => users.push(data))
+
+    .on('end', async () => {
+      console.log(users);
+      console.log(`Processed ${users.length} successfully`);
+
+      users.forEach(async (user) => {
+        let userRole: { id: number; name: string };
+
+        const generatedRole = generateRole(user.role, user.place);
+
+        const roleExists = rolesAvailableOnSuperset.find(
```

Comment:
The const name `roleExists` implies a boolean. Perhaps rename it to something more descriptive.